### PR TITLE
chore: Adds eslint hook plugin and fixes hooks errors

### PR DESCRIPTION
**Pull request by Ubax**
## Description

Adds plugin for checking hooks

## Checklist

- [x] Ensured that CI passes

### CODE REVIEW

File: src/reanimated/ReanimatedScreen.tsx [hunk 1]
```diff
@@ -1,103 +1,29 @@
...
-                [
-                  // This should not be necessary, but is not properly managed by `react-native-reanimated`
-                  // @ts-ignore wrong type
-                  Platform.OS === 'android'
-                    ? 'onTransitionProgress'
-                    : 'topTransitionProgress',
-                ]
-              )
-        }
+        onTransitionProgressReanimated={undefined}
```
Comment (WoLewicki) [inline src/reanimated/ReanimatedScreen.tsx:19]: I think you can just not set this prop at all.

File: src/reanimated/ReanimatedScreen.tsx [hunk 2]
```diff
@@ -1,103 +1,29 @@
...
         {...rest}>
```
Comment (WoLewicki) [inline src/reanimated/ReanimatedScreen.tsx:20]: You probably can just do `...props` here and not add `{children}` explicitly.

**Review by WoLewicki — COMMENTED** [review]
I added some comments

**Review by kacperkapusciak — APPROVED** [review]

File: src/reanimated/ReanimatedScreen.tsx [hunk 3]
```diff
@@ -1,103 +1,25 @@
...
   (props, ref) => {
-    const { children, ...rest } = props;
-
-    const progress = useSharedValue(0);
-    const closing = useSharedValue(0);
-    const goingForward = useSharedValue(0);
-
     return (
       <AnimatedScreen
         // @ts-ignore some problems with ref and onTransitionProgressReanimated being "fake" prop for parsing of `useEvent` return value
```
Comment (WoLewicki) [inline src/reanimated/ReanimatedScreen.tsx:15]: Could you check if those `@ts-ignore` are still needed? Maybe we can get rid of them now.

File: src/reanimated/ReanimatedScreenProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  render() {
+    const ReanimatedScreen = this.props.isNativeStack
+      ? ReanimatedNativeStackScreen
+      : AnimatedScreen;
+    return (
+      <ReanimatedScreen
+        {...this.props}
+        // @ts-ignore some problems with ref
```
Comment (WoLewicki) [inline src/reanimated/ReanimatedScreenProvider.tsx:26]: Same here.

**Review by WoLewicki — COMMENTED** [review]
I added 2 comments, if they cannot be resolved, it is ready to be merged.

File: src/reanimated/ReanimatedScreen.tsx (hunk 3 above)
Comment (Ubax) [reply src/reanimated/ReanimatedScreen.tsx:15]: It is still required

File: src/reanimated/ReanimatedScreenProvider.tsx (hunk 4 above)
Comment (Ubax) [reply src/reanimated/ReanimatedScreenProvider.tsx:26]: Same here :/

**Review by WoLewicki — APPROVED** [review]

**Pull request merged**